src/overlays/actors/ovl_En_Dai/z_en_dai.c

```diff
@@ -22,58 +43,697 @@ const ActorInit En_Dai_InitVars = {
     (ActorFunc)EnDai_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3DFF0.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E168.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E460.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E5B4.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E5DC.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E69C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E7C8.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Dai_0x80B3DFF0/func_80B3E834.asm")
+static Vec3f D_80B3FBF0 = { 1.0f, 1.0f, 1.0f };
+
+EnDaiParticle* func_80B3DFF0(EnDaiParticle* particle, Vec3f arg1, Vec3f arg2, Vec3f arg3, f32 arg4, f32 arg5,
+                             s32 arg6) {
+    s32 i;
+
+    for (i = 0; i < 32; i++, particle++) {
+        if (!particle->isEnabled) {
+            particle->isEnabled = true;
+            particle->unk_01 = (Rand_ZeroOne() * (2.0f * (arg6 / 3.0f))) + (arg6 / 3.0f);
+            particle->unk_02 = particle->unk_01;
+            particle->unk_10 = arg1;
+            particle->unk_1C = arg2;
+            particle->unk_28 = arg3;
+            particle->unk_34 = arg4;
+            particle->unk_38 = arg5;
+            return particle;
+        }
+    }
+    return NULL;
+}
+
+void func_80B3E168(EnDaiParticle* particle, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    s32 pad;
+    s32 flag = 0;
+    s32 i;
+    f32 alpha;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    for (i = 0; i < 32; i++, particle++) {
+        if (particle->isEnabled == true) {
+            gDPPipeSync(POLY_XLU_DISP++);
+
+            if (!flag) {
```

Comment:
Not a huge concern, but a more descriptive name than `flag` would be nice here (i.e. what is this flag tracking? Seems like it's set to true the first time `particle->isEnabled` is true)